compiler/rustc_resolve/src/check_unused.rs

```diff
@@ -315,12 +316,20 @@ impl Resolver<'_> {
                 "remove the unused import"
             };
 
+            let parent_module = visitor.r.get_nearest_non_block_module(
+                visitor.r.local_def_id(unused.use_tree_id).to_def_id(),
+            );
+            let test_module_span = match module_to_string(parent_module) {
+                Some(module) if module.contains("test") => Some(parent_module.span),
```

Comment:
@TaKO8Ki @wesleywiser

If my understanding is correct, This would detect modules like the following as 'test' module.

```rust
mod fastest_search {
  // ...
}
```

Is it intentional? I'm afraid that this loose check causes many false positives. I'd recommend to check more strictly like `module == "test" || module == "tests"`.

Reply:
@wesleywiser I fixed this false positive. Could you review again?

Reply:
As of c597dc8, it looks good to me :+1: